PythonScript/src/ConsoleDialog.cpp

```diff
@@ -609,35 +608,18 @@ void ConsoleDialog::writeText(size_t length, const char *text)
 void ConsoleDialog::writeColoredText(size_t length, const char *text)
 {
 	size_t docLength = (size_t)callScintilla(SCI_GETLENGTH);
-	size_t realLength = length;
 	callScintilla(SCI_SETREADONLY, 0);
-	for (idx_t i = 0; i < length; ++i)

```

Comment:
Why is this no longer needed?

Reply:
Because the complete text should be colored - no need for special treatments like 
in writeerror function but to be honest, I don't see why this special treamtent is needed at all or
what the benefit is.

Reply:
Added with:

Revision: d725887d923dd5d0dcb948d4002038baf0a7333e
Author: Dave Brotherstone <davegb@pobox.com>
Date: 13.08.2010 00:00:27
Message:
Stripped the \r from console output

 - console.run output now a bit nicer if commands output \r\n
----
Modified: PythonScript/src/ConsoleDialog.cpp

Reply:
but I don't see the necessity. 
Example with console.write and print. Note, write mehtod does not, contrary to print,
add an eol automatically to the end of the text.

```
>>> console.write('hello\r')
hello
>>> console.write('hello\n')
hello
>>> console.write('hello\r\n')
hello
>>> print('hello\r')
hello
>>> print('hello\n')
hello

>>> print('hello\r\n')
hello

>>>
```

![grafik](https://user-images.githubusercontent.com/16170444/42940469-d7e6dea2-8b59-11e8-936a-31cbb5ce766a.png)


The different behavior of print is related to python as it is internally using \\n only, that
is the reason why the 2nd and third are having an additional line - but I don't think that
this was the reason to introduce this logis at all.
Coult it be that scintilla from 2010 had problems with different EOLs?
I did a search on scintilla google groups but I haven't found one.
Today, clearly, it hasn't. I can reintroudce the code, I don't mind - Just not sure if it is needed anymore. What about keeping it for some time and if no one complains remove the logic from the 
others as well?
As said, I don't mind, if I should keep it the same like the others I will do so.

Reply:
Seems to be ok, let's see if someone finds an issue with it